service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/CloneControlledAzureDatabaseResourceFlightTest.java

```diff
@@ -0,0 +1,168 @@
+            .managedIdentityName("idsource")
+            .build();
+
+    azureUtils.createResource(
+        sourceWorkspace.workspaceId(),
+        userRequest,
+        sourceManagedIdentityResource,
+        WsmResourceType.CONTROLLED_AZURE_MANAGED_IDENTITY,
+        managedIdentityCreationParameters);
+
+    // Set up a test "source" database
+    var databaseName = "workflowclonetest";
+
+    var creationParameters =
+        ControlledAzureResourceFixtures.getAzureDatabaseCreationParameters(databaseName, false);
+
+    var sourceDatabaseResource =
+        ControlledAzureResourceFixtures.makeSharedControlledAzureDatabaseResourceBuilder(
+                creationParameters, sourceWorkspace.workspaceId())
+            .databaseOwner(sourceManagedIdentityResource.getWsmResourceFields().getName())
+            .databaseName(databaseName)
+            .build();
+
+    azureUtils.createResource(
+        sourceWorkspace.workspaceId(),
+        userRequest,
+        sourceDatabaseResource,
+        WsmResourceType.CONTROLLED_AZURE_DATABASE,
+        creationParameters);
+
+    // Set up a destination managed identity,
+    // simulating the result of the "clone managed identity" flight.
+    var destinationManagedIdentityResource =
+        ControlledAzureResourceFixtures.makeDefaultControlledAzureManagedIdentityResourceBuilder(
+                managedIdentityCreationParameters, destinationWorkspace.workspaceId())
+            .managedIdentityName("iddestination")
+            .build();
+
+    azureUtils.createResource(
+        destinationWorkspace.workspaceId(),
+        userRequest,
+        destinationManagedIdentityResource,
+        WsmResourceType.CONTROLLED_AZURE_MANAGED_IDENTITY,
+        managedIdentityCreationParameters);
+
+    inputs.put(WorkspaceFlightMapKeys.ResourceKeys.RESOURCE, sourceDatabaseResource);
+    inputs.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
+
+    inputs.put(
+        WorkspaceFlightMapKeys.ResourceKeys.CLONING_INSTRUCTIONS,
+        CloningInstructions.COPY_RESOURCE.name());
+
+    inputs.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_ID, UUID.randomUUID());
+
+    inputs.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_NAME, databaseName);
+
+    inputs.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID,
+        destinationWorkspace.workspaceId());
+
+    inputs.put(
+        WorkspaceFlightMapKeys.ControlledResourceKeys.CLONED_MANAGED_IDENTITY,
+        destinationManagedIdentityResource.getWsmResourceFields().getName());
+
+    var result =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            CloneControlledAzureDatabaseResourceFlight.class,
+            inputs,
+            Duration.ofMinutes(1),
+            null);
+
+    assertEquals(FlightStatus.SUCCESS, result.getFlightStatus());
+
+    ClonedAzureResource resultDatabase =
```

Comment:
I don't have any assertions here because we shouldn't merge this connected test. The workflow cloning project should have only one connected test in the end - a test that clones an entire workspace and demonstrates that a database has been cloned as part of that process.